analyzer/codechecker_analyzer/cmd/checkers.py

```diff
@@ -202,217 +206,273 @@ def add_arguments_to_parser(parser):
 
-        config_handler.initialize_checkers(context,
-                                           checkers,
-                                           profile_checkers)
+    if rows:
+        print(twodim.to_str(args.output_format, header, rows))
 
-        for checker_name, value in config_handler.checks().items():
-            state, description = value
+    analyzer_types.print_unsupported_analyzers(errored)
 
-            if state != CheckerState.enabled and 'profile' in args:
-                continue
+    if analyzer_failures:
+        LOG.error("Failed to get checker configuration options for '%s' "
+                  "analyzer(s)! Please try to upgrade your analyzer "
+                  "version to use this feature.",
+                  ', '.join(analyzer_failures))
+        sys.exit(1)
 
-            if state == CheckerState.enabled and 'only_disabled' in args:
-                continue
-            elif state != CheckerState.enabled and 'only_enabled' in args:
-                continue
 
-            if args.output_format == 'json':
-                state = state == CheckerState.enabled
-            else:
-                state = '+' if state == CheckerState.enabled else '-'
-
-            if args.guideline is not None:
-                if not match_guideline(checker_name, args.guideline):
-                    continue
-
-            if 'details' in args:
-                severity = context.severity_map.get(checker_name)
-                guideline = context.guideline_map.get(checker_name, {})
-                if args.output_format != 'json':
-                    guideline = format_guideline(guideline)
-                rows.append([state, checker_name, analyzer,
-                             severity, guideline, description])
-            else:
-                rows.append([checker_name])
+def main(args):
+    """
+    List the checkers available in the specified (or all supported) analyzers
+    alongside with their description or enabled status in various formats.
+    """
 
-    if 'show_warnings' in args:
-        severity = context.severity_map.get('clang-diagnostic-')
-        for warning in get_warnings(analyzer_environment):
-            warning = 'clang-diagnostic-' + warning
+    # If the given output format is not 'table', redirect logger's output to
+    # the stderr.
+    logger.setup_logger(args.verbose if 'verbose' in args else None,
+                        None if args.output_format == 'table' else 'stderr')
 
-            if args.guideline is not None:
-                if not match_guideline(warning, args.guideline):
-                    continue
+    cl = analyzer_context.get_context().checker_labels
 
-            guideline = context.guideline_map.get(warning, {})
-            if args.output_format != 'json':
-                guideline = format_guideline(guideline)
+    # TODO: --profile is a deprecated flag. This section should be removed in
+    # the next release.
```

Comment:
If this flag is depreated, please identify it in the help message too, that this flag is deprecated. Also please print a warning message if this flag is used and tell the user that this flag is deprecated and which command needs to be used instead of this.

By the way if we deprecate this option how will you able to list the profiles (CodeChecker checkers --profile list)?

The following command doesn't work:
```
$ CodeChecker checkers --label profile:list
[ERROR 2021-08-23 13:22] - 'list' is a reserved profile keyword. 
[ERROR 2021-08-23 13:22] - Please choose another profile name in ~/codechecker/build/CodeChecker/config/checker_labels.json and rebuild.
```

Reply:
I removed this comment because we'll not deprecate this flag. I also fixed that command.